openassessment/xblock/static/js/src/studio/oa_edit_settings.js

```diff
@@ -27,11 +28,38 @@ OpenAssessment.EditSettingsView = function(element, assessmentViews) {
         "#openassessment_submission_due_time"
     ).install();
 
+    new OpenAssessment.SelectControl(
+        $("#openassessment_submission_upload_selector", this.element),
+        {'custom': $("#openassessment_submission_white_listed_file_types_wrapper", this.element)},
+        new OpenAssessment.Notifier([
+            new OpenAssessment.AssessmentToggleListener()
+        ])
+    ).install();
+
     this.leaderboardIntField = new OpenAssessment.IntField(
         $("#openassessment_leaderboard_editor", this.element),
         { min: 0, max: 100 }
     );
 
+    this.fileTypeWhiteListInputField = new OpenAssessment.InputControl(
+        $("#openassessment_submission_white_listed_file_types", this.element),
+        function(value) {
+            var badExts = [];
+            var errors = [];
+            var whiteList = $.map(value.split(','), $.trim);
+            $.each(whiteList, function(index, ext) {
+                if (data.FILE_EXT_BLACK_LIST.indexOf(ext) !== -1) {
+                    badExts.push(ext);
+                }
+            });
+            if (badExts.length) {
+                errors.push(gettext('The following file types are not allowed: ') + badExts.join(','));
+            }
```

Comment:
@lamagnifica the missing ".exe" is because it has period and it doesn't match anything in the black list. I can fix it by striping the leading periods in from the input.

To answer your question:
1. No, it is not case sensitive. The extensions are changed to lower case to compare against the blacklist or whitelist. So really_bad.DMG will be blocked.
2. Because 1, we don't need uppercase version
3. Since the filenames are split by period, there will be no period in the extension. So we don't need to include period in blacklist.